pkg/controller/datavolume-controller.go

```diff
@@ -515,6 +697,278 @@ func (r *DatavolumeReconciler) sourceInUse(dv *cdiv1.DataVolume) (bool, error) {
 	return len(pods) > 0, nil
 }
 
+func (r *DatavolumeReconciler) initTransfer(dv *cdiv1.DataVolume, name string) (bool, error) {
+	initialized := true
+
+	if !HasFinalizer(dv, crossNamespaceFinalizer) {
+		AddFinalizer(dv, crossNamespaceFinalizer)
+		if err := r.client.Update(context.TODO(), dv); dv != nil {
+			return false, err
+		}
+
+		initialized = false
+	}
+
+	ot := &cdiv1.ObjectTransfer{}
+	if err := r.client.Get(context.TODO(), types.NamespacedName{Name: name}, ot); err != nil {
+		if !k8serrors.IsNotFound(err) {
+			return false, err
+		}
+
+		if err := validateCloneTokenDV(r.tokenValidator, dv); err != nil {
+			return false, err
+		}
+
+		ot = &cdiv1.ObjectTransfer{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: name,
+			},
+			Spec: cdiv1.ObjectTransferSpec{
+				Source: cdiv1.TransferSource{
+					Kind:      "PersistentVolumeClaim",
+					Namespace: dv.Spec.Source.PVC.Namespace,
+					Name:      name,
+					RequiredAnnotations: map[string]string{
+						annReadyForTransfer: "true",
+					},
+				},
+				Target: cdiv1.TransferTarget{
+					Namespace: &dv.Namespace,
+					Name:      &dv.Name,
+				},
+			},
+		}
+
+		if err := setAnnOwnedByDataVolume(ot, dv); err != nil {
+			return false, err
+		}
+
+		if err := r.client.Create(context.TODO(), ot); err != nil {
+			return false, err
+		}
+
+		initialized = false
+	}
+
+	return initialized, nil
+}
+
+func (r *DatavolumeReconciler) cleanupTransfer(log logr.Logger, dv *cdiv1.DataVolume, name string) error {
+	if !HasFinalizer(dv, crossNamespaceFinalizer) {
+		return nil
+	}
+
+	log.Info("Doing cleanup")
+
+	if dv.DeletionTimestamp != nil && dv.Status.Phase != cdiv1.Succeeded {
+		// delete all potential PVCs that may not have owner refs
+		namespaces := []string{dv.Namespace}
+		names := []string{dv.Name}
+		if dv.Spec.Source.PVC != nil &&
+			dv.Spec.Source.PVC.Namespace != "" &&
```

Comment:
So I think this is going to clash with the dv.Spec.Storage stuff Bartosz is doing.

Reply:
I have updated my PR with a commit `Refactor: Render the effective PVC early`. This should help integrate the two PR's

Reply:
here: https://github.com/kubevirt/containerized-data-importer/pull/1753

Reply:
my latest change creates a `pvcSpec`, so if you use actual pvc Spec instead of `dv.Spec.PVC` then integration would be easy. 

Actually I think it would not apply here to code on line 768, becasue this is source pvc `dv.Spec.Source.PVC` which is not affected. But line 842 might actually be in conflict.